app/components/offering-editor.js

```diff
@@ -0,0 +1,190 @@
+        let exp = new RegExp(filter, 'gi');
+
+        let activeGroupFilter = function(learnerGroup) {
+          let searchTerm = `${learnerGroup.get('title')}${learnerGroup.get('allParentsTitle')}`;
+
+          return (
+            learnerGroup.get('title') !== undefined &&
+            proxy.get('selectedLearnerGroups') &&
+            exp.test(searchTerm) &&
+            !proxy.get('selectedLearnerGroups').contains(learnerGroup)
+          );
+        };
+
+        this.get('content.topLevelLearnerGroups').then((cohortGroups) => {
+          let learnerGroups = [];
+          let promises = [];
+
+          cohortGroups.forEach((learnerGroup) => {
+            learnerGroups.pushObject(learnerGroup);
+
+            let promise = new Promise((resolve) => {
+              learnerGroup.get('allDescendants').then((descendants) => {
+                learnerGroups.pushObjects(descendants);
+                resolve();
+              });
+            });
+
+            promises.pushObject(promise);
+          });
+
+          all(promises).then(() => {
+            defer.resolve(learnerGroups.filter(activeGroupFilter).sortBy('sortTitle'));
+          });
+        });
+
+        return PromiseArray.create({
+          promise: defer.promise
+        });
+      }),
+    });
+
+    let cohorts = this.get('cohorts') ? this.get('cohorts') : [];
+
+    return cohorts.map((cohort) => {
+      let proxy = cohortProxy.create({
+        content: cohort,
+        selectedLearnerGroups: this.get('learnerGroups')
+      });
+
+      return proxy;
+    }).sortBy('title');
+  }),
+
+  actions: {
+    setOfferingType(value) {
+      this.set('singleOffering', value);
+    },
+
+    toggleMultiDay() {
+      this.set('isMultiDay', !this.get('isMultiDay'));
+    },
+
+    changeStartTime(date) {
+      let newStart = moment(date);
+      let startDate = moment(this.get('startDate'));
+
+      startDate.hour(newStart.format('HH'));
+      startDate.minute(newStart.format('mm'));
+      this.set('startDate', startDate.toDate());
+    },
+
+    changeEndTime(date) {
+      let newEnd = moment(date);
+      let endDate = moment(this.get('endDate'));
+
+      endDate.hour(newEnd.format('HH'));
+      endDate.minute(newEnd.format('mm'));
+      this.set('endDate', endDate.toDate());
+    },
+
+    addInstructorToBuffer(instructor){
+      this.get('instructors').pushObject(instructor);
+    },
+
+    addInstructorGroupToBuffer(instructorGroup) {
+      this.get('instructorGroups').pushObject(instructorGroup);
+    },
+
+    removeInstructorFromBuffer(instructor){
+      this.get('instructors').removeObject(instructor);
+    },
+
+    removeInstructorGroupFromBuffer(instructorGroup){
+      this.get('instructorGroups').removeObject(instructorGroup);
+    },
+
+    addLearnerGroup(group) {
+      let learnerGroups = this.get('learnerGroups');
+
+      group.get('allDescendants').then((descendants) => {
+        if (isEmpty(descendants)) {
```

Comment:
This is good code, but it needs a comment along the lines of 
//If the group has no descendants then we add it, otherwise we add the descendants, but not the parent.